### PR TITLE
fix: flux error properly read by cloud

### LINKED ISSUE #22327: Flux disabled error message should be a flux-formatted error
Currently we return a plaintext string when flux is disabled in 1.x . This results in bad error messages from tools that speak flux, e.g. this gem from :

`Error exhausting result iterator; Err: invalid character 'F' looking for beginning of value: invalid character 'F' looking for beginning of value`

We should return a flux-native error message instead.

### PR DESCRIPTION
**Pull request by lesam**
Closes #22327 

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by lesam** [comment]
Before this change:

Screenshot 2021-08-31 at 10 33 06 AM

After this change:

Screenshot 2021-08-31 at 10 33 16 AM

File: tests/server_test.go [hunk 1]
```diff
@@ -10237,6 +10237,31 @@ func TestFluxRegressionEndToEnd(t *testing.T) {
...
+	query.Type = "flux"
+	query.Dialect.Annotations = csv.DefaultDialect().Annotations
+	body, err := json.Marshal(query)
+	require.NoError(t, err)
+	resp, err := http.Post(u.String(), "application/json", bytes.NewBuffer(body))
+	require.NoError(t, err)
+	assert.Equal(t, 403, resp.StatusCode)
+	body, err = ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	assert.Equal(t,  `{"error":"Flux query service disabled. Verify flux-enabled=true in the [http] section of the InfluxDB config."}` + "\n", string(body))
```
Comment (lesam) [inline tests/server_test.go:10262]: Previously this was plaintext instead of json

File: tsdb/engine/tsm1/engine_test.go [hunk 2]
```diff
@@ -2873,12 +2873,12 @@ func MustParsePointString(buf string) models.Point { return MustParsePointsStrin
 type mockPlanner struct{}
 
 func (m *mockPlanner) Plan(lastWrite time.Time) ([]tsm1.CompactionGroup, int64) { return nil, 0 }
-func (m *mockPlanner) PlanLevel(level int) ([]tsm1.CompactionGroup, int64) { return nil, 0 }
-func (m *mockPlanner) PlanOptimize() ([]tsm1.CompactionGroup, int64)       { return nil, 0 }
-func (m *mockPlanner) Release(groups []tsm1.CompactionGroup)               {}
-func (m *mockPlanner) FullyCompacted() (bool, string)                  { return false, "not compacted" }
-func (m *mockPlanner) ForceFull()                                      {}
-func (m *mockPlanner) SetFileStore(fs *tsm1.FileStore)                 {}
+func (m *mockPlanner) PlanLevel(level int) ([]tsm1.CompactionGroup, int64)      { return nil, 0 }
```
Comment (lesam) [inline tsdb/engine/tsm1/engine_test.go:2876]: Looks like our formatting checks have some gaps...

File: .circleci/config.yml [hunk 3]
```diff
@@ -94,8 +94,11 @@ jobs:
             set -x
             mkdir -p junit
             go test -v ./... 2>&1 | tee tests.log
-            go-junit-report < tests.log > junit/influxdb.junit.xml
           no_output_timeout: 1500s
+      - run:
+          name: Convert test output to junit
+          command: go-junit-report < tests.log > junit/influxdb.junit.xml
```
Comment (lesam) [inline .circleci/config.yml:100]: Fix to always report which tests actually failed (especially when a test actually does fail).

**Review by gwossum — APPROVED** [review]
Looks good to me. Pretty subtle change to get it to output JSON

**Pull request merged**